tests/run-jbang-tests.sh

```diff
@@ -0,0 +1,339 @@
+#!/bin/bash
+
+script_dir=$(dirname "$0")
+
+executionMode=$1
+
+test_failure=0
+test_count=0
+TEST_TIMEOUT=20
+
+test_dir=tests/target/jbang/
+
+glow_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+
+function setupTestSuite {
+  echo Clear JBang cache
+  jbangExec cache clear
+  mkdir -p $test_dir
+}
+
+function tearDownTestSuite {
+  rm -rf $test_dir
+}
+
+function check_endpoint() {
+    local url=$1
+
+    response=$(curl -s -w "%{http_code}" $url)
+    echo "Request responded with : $response" >&2
+    http_code="${response: -3}"
+    length=${#response}
+    body="${response:0:length-3}"
+
+    if [[ "$http_code" == "200" ]]; then
+        echo "$body"
+        return 0
+    else
+        echo "Request failed with status: $http_code" >&2
+        return 1
+    fi
+}
+
+function check_endpoint_response() {
+    local url=$1
+    local expected_body=$2
+
+    body=$(check_endpoint $url)
+    if [[ $? -eq 0 ]]; then
+      if [[ "${body}" == "${expected_body}" ]]; then
+        return 0
+      else
+        echo Got: ${body} but expected: ${expected_body} >&2
+        return 1
+      fi
+    else
+      return 1
+    fi
+}
+
+function jbangExec {
+  curl -Ls https://sh.jbang.dev | bash -s - $@
+}
+
+function jbangBuild {
+  local java_source_file=$1
+  echo "Building with source"
+  cat $java_source_file
+
+  jbangExec --verbose build $java_source_file
+  if [ $? -ne 0 ]; then
+    echo "Error building the application with JBang, check log"
+    test_failure=1
+    return 1
+  fi
+  return 0
+}
+
+function jbangRun {
+  local java_source_file=$1
+  local verifier=$2
+
+  # Run the application and verify it works as expected
+  jbangExec --verbose run $java_source_file &
+
+  failed=0
+  elapsed=0
+  sleep 5
+  while ! eval ${verifier}; do
+    sleep 1
+    ((elapsed++))
+
+    if [[ $elapsed -ge $TIMEOUT ]]; then
+        echo "Timeout reached: Process did not start responding on port 8080 within $TEST_TIMEOUT seconds."
+        test_failure=1
+        failed=1
+        break
+    fi
+  done
+  wildfly_pid=$(lsof -ti :8080)
+  echo "Kill WildFly process (${wildfly_pid})"
+  kill $wildfly_pid
+  return $failed
+}
+
+# Test a simple Jakarta RS application
+# without any 3rd-party dependencies
+# (all deps are provided by WildFly)
+function testApp {
+  test_count=$((test_count+1))
+  java_source_file=$test_dir/app.java
+  cat <<- EOF > $java_source_file
+///usr/bin/env jbang "$0" "$@" ; exit $?
+//JAVA 17+
+//DEPS org.wildfly.bom:wildfly-expansion:35.0.1.Final@pom
```

Comment:
@jfdenise something I could improve it find the latest server version and use it in the tests instead of hard-code version that will quickly be outdated. wdyt?

Reply:
done in bf35ff29c3dc39645a8aad78997c19c448a74774